code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -667,28 +667,41 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 			if(.)
 				release_pressure = clamp(round(pressure), can_min_release_pressure, can_max_release_pressure)
 				investigate_log("was set to [release_pressure] kPa by [key_name(usr)].", INVESTIGATE_ATMOS)
-		if("valve")
+		if("valve")		//logging for openning canisters
 			var/logmsg
 			valve_open = !valve_open
 			if(valve_open)
 				SSair.start_processing_machine(src)
 				logmsg = "Valve was <b>opened</b> by [key_name(usr)], starting a transfer into \the [holding || "air"].<br>"
 				if(!holding)
 					var/list/danger = list()
+					///list for logging all gases in canister
+					var/list/gaseslog = list()		
 					for(var/id in air_contents.gases)
 						var/gas = air_contents.gases[id]
+						gaseslog[gas[GAS_META][META_GAS_NAME]] = gas[MOLES] 
 						if(!gas[GAS_META][META_GAS_DANGER])
 							continue
 						if(gas[MOLES] > (gas[GAS_META][META_GAS_MOLES_VISIBLE] || MOLES_GAS_VISIBLE)) //if moles_visible is undefined, default to default visibility
 							danger[gas[GAS_META][META_GAS_NAME]] = gas[MOLES] //ex. "plasma" = 20
-
+					/**
+					* just works better to use danger
+					* list as a test for admin alerts
+					*/
+					///alert + log path
 					if(danger.len)
 						message_admins("[ADMIN_LOOKUPFLW(usr)] opened a canister that contains the following at [ADMIN_VERBOSEJMP(src)]:")
 						log_admin("[key_name(usr)] opened a canister that contains the following at [AREACOORD(src)]:")
-						for(var/name in danger)
-							var/msg = "[name]: [danger[name]] moles."
+						for(var/name in gaseslog)
+							var/msg = "[name]: [gaseslog[name]] moles."
 							log_admin(msg)
 							message_admins(msg)
+					///just logging, no alert
+					else
+						log_admin("[key_name(usr)] opened a canister that contains the following at [AREACOORD(src)]:")
+						for(var/name in gaseslog)
+							var/msg = "[name]: [gaseslog[name]] moles."
+							log_admin(msg)
```

Comment:
Somewhat in the same vein, these two for loops are aggressively similar. instead of using an if statement to choose between them, why not have checks for tell_admins around the message admins part, and merge the two loops into one?